docs/NNNN-direct-clean-storage-layout.md

```diff
@@ -0,0 +1,99 @@
+# OCFL Community Extension NNNN: Direct Clean Storage Layout
+
+* **Extension Name:** NNNN-direct-clean-storage-layout
+* **Authors:** Jürgen Enge, Basel
+* **Minimum OCFL Version:** 1.0
+* **OCFL Community Extensions Version:** 1.0
+* **Obsoletes:** n/a
+* **Obsoleted by:** n/a
+
+## Overview
+
+This object/storage root extension maps content file names or OCFL object identifiers by replacing "dangerous characters" from file and folder names.
+
+The functionality derived from David A. Wheeler - "Fixing Unix/Linux/POSIX Filenames" (https://www.dwheeler.com/essays/fixing-unix-linux-filenames.html)
+
+## Parameters
+
+### Summary
+
+* **Name:** `maxLen`
+   * **Description:** determines the maximum number of result characters.
+     The default value is `255`, which means that a result with more than 255 characters will raise an error.
+    * **Type:** number
+   * **Constraints:** An integer greater 0
+   * **Default:** 255
+  
+## Procedure
+
+The following is an outline to the steps for mapping an identifier/filepath:
+
+1. Replace all non-UTF8 characters with "_"
+2. Split the string at path separator "/"
+3. For each part do the following
+   1. Replace any character from this list with "_": 0x00-0x1f 0x7f * ? : [ ] " <> | ( ) { } & ' ! ; # @
+   2. Remove leading spaces, "-" and "~" / remove trailing spaces
```

Comment:
Path parts that are exclusively a series of periods are also problematic (eg `.` or `..`).